### DISCUSSION TITLE
Webhooks?

### DISCUSSION THREAD

**Discussion by corysimmons** [payloadcms/payload#1081] [Q&A] [answered]
Does Payload have any concept of webhooks on certain events?

**Comment by DanRibbens** [comment, marked as answer]
Payload doesn't have webhooks, it gives you a complete API that other apps can call. You can extend your Payload app to call any code by adding your own hooks https://payloadcms.com/docs/hooks/overview.

You can also add custom endpoints through payload or on your express app that can listen for calls.

What are you trying to do?

**Reply by corysimmons** [reply]
Thank you.

As an example, in my old CMS, anytime I would publish an Article, I'd have a webhook set up that hit a custom endpoint that recorded some audit log stuff to a table.

Just _lots_ of little things like that where if a piece of content is mutated, I can send the payload of that mutation to a specific endpoint.

**Reply by jmikrut** [reply]
You can do all that with Payload Hooks - they can do everything webhooks can do but WAY better!

**Reply by corysimmons** [reply]
Awesome, awesome. Thank you! :)
